net/inet_pton.c

```diff
@@ -1,6 +1,147 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * net/inet_pton.c
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
 #include <arpa/inet.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
 
-int inet_pton(int af, const char *restrict src, void *restrict dst)
+/* Accept only addresses in dotted-decimal format ddd.ddd.ddd.ddd */
+static int inet_pton4(const char *src, struct in_addr *dst)
 {
-	return 0;
+	in_addr_t addr4 = 0;
+	unsigned long val;
+	int i;
+	char *endptr;
+
+	for (i = 0; i < 4; i++) {
+		if (!isdigit(*src))
+			return 0;
+
+		val = strtoul(src, &endptr, 10);
```

Comment:
As for me checking if the first character is a digit is redundant, function `strtoul()`  checks it anyway.

Reply:
According to [Open Group](https://pubs.opengroup.org/onlinepubs/009604499/functions/strtoul.html) `strtoul()` accepts leading white spaces as well as a sign character. I was referring to that documentation, when writing my function. I have just checked that our implementation of `strtoul` does not conform to that documentation and won't accept such strings. However, I think that this one redundant check is not that painful, given that someone might fix the `strtoul` implementation in the future.